lib/Conversion/TorchToLinalg/Uncategorized.cpp

```diff
@@ -354,6 +354,30 @@ static Value createLinalgPayloadCalculationForElementwiseOp(
     Value cdfExt = b.create<arith::AddFOp>(loc, dinputInputAlpha, cdf);
     return b.create<arith::MulFOp>(loc, payloadArgs[0], cdfExt);
   }
+  if (auto hardtanhBackward = dyn_cast<AtenHardtanhBackwardOp>(op)) {
+    AtenHardtanhBackwardOp::Adaptor adaptor(operands);
+    if (!hardtanhBackward.getType()
+             .cast<ValueTensorType>()
+             .getDtype()
+             .isa<mlir::FloatType>()) {
+      hardtanhBackward.emitError("unimplemented: non-floating point dtype");
+      return nullptr;
+    }
+    Type elementType = payloadArgs[0].getType();
+    Value gradOutput =
+        convertScalarToDtype(b, loc, payloadArgs[0], elementType);
```

Comment:
This type conversion is not needed, since `elementtType` is the type of `payloadArgs[0]`